<html>

<head>
<style>
body{color:black;font-family:"Verdana", "Arial", "Helvetica", "sans-serif";font-size:9pt;margin:0}
td{color:black;font-family:"Verdana", "Arial", "Helvetica", "sans-serif";font-size:9pt}
a:link       { text-decoration: none; color:#0000CC }
a:visited    { text-decoration: none; color: #0000CC }a:hover      { color: #0000CC }
</style>
<style type="text/css">
body{margin-left:60pt;}
body{margin-right:60pt;}
body{text-align: justify;}
</style>
<title>Orthofuzz</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8”>  
</head>

<body>

<p align="center">
<br>
<table border="0" cellpadding="0" cellspacing="0" style="border-collapse: collapse" bordercolor="#111111" width="100%" id="AutoNumber1" height="83">
  <tr>
    <td rowspan="4" height="83">
    <a href="http://orthofuzz.github.io/">
    <img border="0" width="150" height="134" src="titlefigure.png"></a></td>
    <td height="15"><font color="#828282">
    <font face="Garamond" color="#000000" size="6"><i>ORTHOFUZZ</font><font face="Garamond" size="6"><b><i>&nbsp;&nbsp;&nbsp;
    </b></i></font></font></td>
  </tr>
  <tr>
    <td width="82%" height="28"><font face="Garamond" color="#000000" size="5"><b>A java library for functional annotation and pathway analysis of denovo assembled 
RNA-Seq data
</b></font></td>
  </tr>
  
 <tr>
    <td width="82%" height="28"><font face="Garamond" color="#000000" size="4"><b>http://orthofuzz.github.io/</b></font></td>
  </tr>
  <tr>
    <td width="82%" height="28"><font face="Garamond" color="#000000" size="4"><b>Nikolau Group - BBMB/BCB - Iowa State University</b></font></td>
  </tr>
</table>
<hr color="#C8C8C8">
<table cellpadding="20"><tr><td>
<p>
<div align="justify">


<span lang="EN-US" style="font-size: 14.0pt; font-family: Garamond">
<ul>
<li>


<p><b>Overview:</b></p>



<p>The Orthofuzz package positions contigs assembled de-novo from RNA-Seq short read data among protein sequences obtained from phylogenetic neighbors of the target organism in a muti-organism pairwise sequence similarity network. </p><p>A set of well annotated organisms in the phylogenetic neighborhood of the target organisms are handpicked and an All vs All BLAST carried out on translated contigs and amino-acid sequences to define an All vs All pairwise sequence similarity network with the sequences forming the nodes and the BLAST bit-score used to define the edge weight (higher the bit-score, higher the homology). </p>
<p>The package includes a series of java programs which uses a combination of network topology and edge weights to annotate, analyze and summarize the RNA-Seq data.</p>


 <h3>Where does Orthofuzz fit into the RNA-Seq Bioinformatics Pipeline ?</h3>

<a href="images/pipeline.png">
    <p style="text-align:center"><img alt="Link not working" src="images/pipeline.png" width="600" height="500"/>
</a>


                                                                                  
<p><b>Key terms used in the descriptions:</b> </p>

<p><b>source organisms</b>: organisms, the annotations of which are used as sources of functional annotations.
<b>source sequences</b>: sequences from the source organisms.
<b>target organism</b>: the denovo sequenced organism, which we are analyzing.
<b>target sequences</b>: sequences from target organisms.
<b>G</b>: The All-vs-All Pairwise Sequence Similarity Network.
<b>Node</b>: A sequence in the pairwise-sequence similarity network.

</p>


<p><b>List of External Packages used: </b></p>
<a href="http://jung.sourceforge.net"> JUNG </a><a href="https://github.com/rwl/ParallelColt"> Paralellcolt </a>
<a href="http://trove.starlight-systems.com"> Trove </a>
<a href="http://jblas.org"> Jblas </a>
<a href="http://www.paccanarolab.org/static_content/clusterone/cl1-cmdline-1.0.html"> ClusterOne </a><p>


<p><b>Output formats compatible with: </b>
<a href="http://www.cytoscape.org/cy3.html"> Cytoscape3 </a>
<a href="http://www.pathvisio.org/"> Pathvisio </a>
<a href="http://www.paccanarolab.org/clusterone/ "> ClusterOne </a>
<a href="http://genepattern.broadinstitute.org "> Genepattern </a>
</p>                                          


<p><b> Programs in the package:</b> <br/> 
(Click on names in <span style="color:#0000CC">BLUE</span> to get detailed descriptions and download links)</p>


                                            
<p><a href="RandomWalk.html"<b>RandomWalkwithRestart.jar </b></a><br/>
<a href="transferannotations.html"<b>TransferAnnotations.jar </b></a><br/>
<a href="TransferExpression.html"<b>TransferExpression.jar </b></a><br/>
<a href = "PredictInteractions.html"<b>PredictInteractions.jar </b></a>
</p>                  
                              
<p>
Previous work has shown that functional ortholog proteins tends to share more neighbors with higher edge weight when compared to weak homologs
in All-vs-All pairwise sequence similarity networks. Edge weigted random walk with restart is a memory efficient and fast computational method for capturing both shared topological neighborhood and high edge-weight in a [0-1] score, thereby making it a very useful tool for measuring orthology between sequences and/or sets of sequences.      
</p>

<p> Given a set of query nodes and an All-vs-All pairwise sequence similarity network, <a href="RandomWalk.html"<b>RandomWalkwithRestart.jar </b></a> reports a cluster of nodes in the network, which are members of or/and homologous to the query set, provide global and organism normalized scores representing the strength of homology of the node to the query-set and statistical significance obtained by permulation test.</p>                                                   
<p>
    The query-set can have a single node or a set of nodes from the same and/or multiple organisms, whose sequences are in the network. This facilitates querying the network for homologs of gene families with one or more members from different source organisms in the network.
</p>                                                         
                                                         
<p>The max normalized <em>orthoscore</em> [0-1] returned for each member node of the cluster is a strong indicator of functional orthology. The RandomWalkwithRestart, therefore is a very powerful tool for [1] transferring functional informations we know about the query sets to the cluster members or for [2] summarizing information we know about the cluster-members to estimate summary statistics for the query-set.
</p>
                                                         
<p><b>[1] Transferring information from query sets to cluster members: </b>
The standard approach for assigning functional annotations to denovo assembled sequences is based on the observation that there is a high probability for cross organism orthologs to share biological function. Previous work has shown that true orthologs tend to share 
more neighbors with high edge weights. Since orthoscore captures this information, we can use the orthoscores for transferring functional annotations associated with the source sequences in the networks to the target sequences.                                                      </p>

Given the source sequences and their functional annotations, <a href="transferannotations.html"<b>TransferAnnotations.jar </b></a> method will transfers functional annotations from the source sequences to the target sequences.</p> 
                                                                                               
If a gene X is known to interact with gene Y in one/many of the source organisms, <a href = "PredictInteractions.html"<b>PredictInteractions.jar </b></a> can use the RandomWalkwithRestart.jar to identify corresponding orthologs in the target organisms and predict possible contigs/sets of contigs which could be coding for proteins which interacts with each other in the target organisms. It generates output which can for input to clusterOne   which can identify possible interaction complexes in the target organisms.</p>


<p>3. Given mappings to genes in source organisms to enzymes and pathways, <a href="TransferExpression.html"<b>TransferExpression.jar </b></a> method will find homologs of the query sets in the new organisms, summarize gene expression values for each query. The <b>GeneSetEnrichmentStats.jar</b> will generate gene-set statistics using Fisher’s method for combining p-values from gene-sets.</p>

                                                                                                            
                                                                                                            
                                                                                                            
<p>4. Given a list of contigs, <b>GeneListStatistics.jar</b> will generate gene list statistics of lists of contigs to identify functional annotation terms enriched in lists of contigs. The enrichment statistics are estimated using EASE scores, which is a modified form of Fisher Exact Test for enrichment.</p>


<p>6. Better strength to this prediction can be obtained by finding interaction-units that are correlated with each other. <b>ParalellCoexpression.jar </b>Estimates pearson correlation of expression data using paralellColt java package. </p>

<b>7. FindRepresentativeCliques.jar </b>finds clusters in the network, first by finding representative cliques in the network and then running NetworkFlow.jar to find weak members of the clusters.
<h4> The program parameters </h4>
<p><a href="network.flow.html"> Click here to see how to use the programs</a><p>


<p>Overview of the workflow:</p>

<p>1. Preparing the input files </p>

<p> The different input files needed for the programs have to be in <tab> delimited text files in specified formats.</p>

<p><a href="Input.formats.html"> The input file formats are detailed here. </a><p>

<p>2. Run NetworkFlow program to estimate the strength of homology between the source and target sequences.</p>

<p>3. Run the TransferExpression program to estimate the representative expression levels for query gene sets from the complied pathways.</p>

<p>4. Run the TransferAnnotations program to transfer the annotations from the source sequences to the target sequences.</p>

<p>5. Run the TransferInteractions program to estimate evidence for the presence and strength of protein interactions of the source organisms in the target organisms.</p>

<p>6. Run Coexpression program on the expression of the interaction data to identify interaction units that are co-expressed.</p>

<p>10. Run NetworkAND program to identify protein complexes that are interacting and are co-expressed in the target organisms.</p>

<p>11. Run Coexpression program on the pathway expressions to identify enzymes and pathways that are co-expressed.</p>


<p><a href="networkflow.html"> Input formats </a><p>

<p>
<br>
<br>
<br>
<br>
<br>
</p>


<p>Key concepts [Manuscript under preparation]:</p>

<p>1. The fragmented nature and the high identity between sub-groups of the contigs make the current approach of assignment of contigs to the best hits in the source organisms [Table X] error-prone [Figure X]. </p>

<p>2. Protein sequences from the same gene family share more neighbors with higher edge weights compared to those that just share functional domains [Figure X].</p>

<p>3. Random walking with restart on an edge-weighted network captures high edge weights and topological neighborhood to give probabilistic definition of network neighborhood [Figure X].</p>

<p>There are five algorithms at the core of the package [Refer Manuscript]:</p>

<p>1. Random walk with restart on an edge weighted network starting from a set of start nodes.</p>

<p>2. Estimate summary expression value of the topological neighbors of a set of nodes.</p>

<p>3. Transfer functional annotations from source organisms to target organisms.</p>

<p>4. Transfer interactions from source organisms to the target organisms.</p>

<p>5. Modified Tomita Algorithm to identify representative cliques in sequence similarity networks.</p>


<p> Statistics </p>

<p>1. Edge Weighted Random Walk with restart using perambulation statistics</p>

<p>2. Permutation Statistics using max mean statistic</p>

<p>3. EASE Score for gene list </p>

<p>4. Fishers Gene Set Stat</p>

<p>5. Max mean stat for Gene Set Analysis</p>


<p>The software was developed in JAVA on GNU/Linux operating system with support for multithreading. It is free for noncommercial use.</p>

<br><br>
<li><b>Source code:</b><br>

<a href="javascript: void(0)" onclick="window.open('download.php','popup','width=400,height=200,scrollbars=no,resizable=no,toolbar=no,directories=no,location=no,menubar=no,status=no,left=0,top=0'); return false">link here</a>

<br><br>
<li><b>Citation:</b><br>
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21594272">Details of Paper here</a>
<br><br>
<li><b>Contact:</b><br>
Adarsh Jose (ajose<b>AT</b>iastate.edu)<br>

</ul>

 <p style="line-height:150%; font-size:12pt;">
</td>
</tr></table>

</body></html>
