<html>

<head>
<style>
body{color:black;font-family:"Verdana", "Arial", "Helvetica", "sans-serif";font-size:9pt;margin:0}
td{color:black;font-family:"Verdana", "Arial", "Helvetica", "sans-serif";font-size:9pt}
a:link       { text-decoration: none; color:#008080 }
a:visited    { text-decoration: none; color: #008080 }a:hover      { color: #00FFCC }
</style>
<style type="text/css">
body{margin-left:60pt;}
body{margin-right:60pt;}
</style>
<title>NetWalker</title>
<meta http-equiv="Content-Language" content="en-us">
<meta name="GENERATOR" content="Microsoft FrontPage 5.0">
<meta name="ProgId" content="FrontPage.Editor.Document">
<meta http-equiv="Content-Type" content="text/html; charset=windows-1252">
</head>

<body>
<p align="center">
<br>
<table border="0" cellpadding="0" cellspacing="0" style="border-collapse: collapse" bordercolor="#111111" width="100%" id="AutoNumber1" height="83">
  <tr>
    <td rowspan="4" height="83">
    <a href="http://bioinfo.vanderbilt.edu/netwalker">
    <img border="0" width="150" height="134" src="netwalker1.png"></a></td>
    <td height="15"><font color="#828282">
    <font face="Arial Black" size="6">NetWalker</font><font face="Arial" size="6"><b>&nbsp;&nbsp;&nbsp;
    </b></font></font></td>
  </tr>
  <tr>
    <td width="82%" height="28"><font face="Arial" color="#828282"><b><i>http://bioinfo.vanderbilt.edu/netwalker</i></b></font></td>
  </tr>
  <tr>
    <td width="82%" height="28"><font face="Arial" color="#828282"><b><i>Vanderbilt University</i></b></font></td>
  </tr>
</table>
<hr color="ff9900">
<table cellpadding="20"><tr><td>
<p>
<div align="left">

<span lang="EN-US" style="font-size: 12.0pt; font-family: Arial">
<ul>
<li><b>Introduction:</b><br>
<br>
NetWalker is based on the random walk analysis described in Zhang et al (see the citation below). It is similar to GeneWanderer (Kohler et al., Am J Hum Genet, 2008). However, a statistical analysis is implemented to evaluate the random walk results.<br><br> 
The random walk technique exploits the global structure of a network (graph) by simulating the behavior of a random walker on a graph. 
From a starting node, the walker selects a neighbor of it at random and moves to the neighboring node.
Then the walker selects a neighbor of this node at random and moves to it, so on and so forth. 
The sequence of nodes selected this way is a random walk on the graph. 
In a variant of the random walk implemented in NetWalker, the walker may also choose to teleport to the start node with a certain probability, called restart probability.
The restart probability enforces a restriction on how far the random walker can get away from the start node. 
<br><br> 
In the current version of NetWalker, the walker can start from an input list of nodes provided by a user with equal probability. 
The final score of a node in the network is defined as the steady-state probability that the random walker would stay at the node.
To assess the statistical significance of the scores, we let the random walker start from the same number of randomly selected nodes to calculate random scores for each node. 
This process is repeated to generate multiple (e.g. 1000) sets of random scores. 
Then, for each node, a local <i>p</i> value is estimated by comparing the real score to random scores from the same node, and a global <i>p</i> value is estimated by comparing the real score to random scores from all nodes. 
A significant global <i>p</i> value indicates the overall significance of the node with regard to the input nodes, while a significant local <i>p</i> value ensures that the significance is not simply due to network topology. 
<br><br>
The software was developed in C++ on GNU/Linux operating system with support for multithreading. It is free for noncommercial use.
<br><br>
<li><b>Source code:</b><br>

<a href="javascript: void(0)" onclick="window.open('download.php','popup','width=400,height=200,scrollbars=no,resizable=no,toolbar=no,directories=no,location=no,menubar=no,status=no,left=0,top=0'); return false">netwalker-1.0.tar.gz</a>

<br><br>
<li><b>Citation:</b><br>
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21594272">Bing Zhang, Zhiao Shi, Dexter T Duncan, Naresh Prodduturi, Lawrence J Marnett, Daniel C Liebler. Relating protein adduction to gene expression changes: a systems approach. <i>Molecular Biosystems.</i> 7(7):2118-27, 2011</a>
<br><br>
<li><b>Contact:</b><br>
Zhiao Shi (zhiao.shi<b>AT</b>accre.vanderbilt.edu)<br>
Bing Zhang (bing.zhang<b>AT</b>vanderbilt.edu)
</ul>

 <p style="line-height:150%; font-size:12pt;">
</td>
</tr></table>





<br><b>Usage statistics since May 19, 2011</b></p>
<i>

The software download count is <b>1247
</b>and the last date of download: <b>10/03/15</b><br />
<br>
<hr color="#FF9900">
<p style="margin-top: 0; margin-bottom: 0" align="left"><font face="Arial" size="2">
<b> <a href="http://bioinfo.vanderbilt.edu/zhanglab/">Zhang Lab  </a></b>
 | Last updated: 05-19-2011<br>
<br><br> 

</body></html>
