<html>

<head>
<style>
body{color:black;font-family:"Verdana", "Arial", "Helvetica", "sans-serif";font-size:12pt;margin:0}
td{color:black;font-family:"Verdana", "Arial", "Helvetica", "sans-serif";font-size:9pt}
a:link       { text-decoration: none; color:#0000CC }
a:visited    { text-decoration: none; color: #0000CC }a:hover      { color: #0000CC }
</style>
<style type="text/css">
body{margin-left:60pt;}
body{margin-right:60pt;}
body{text-align: justify;}
    
figcaption {
color: green;
}
</style>
<title>Orthofuzz</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8â€> 
   
                                         
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">                                         
</head>

<body>

                                       
<p align="center">
<br>
<table border="0" cellpadding="0" cellspacing="0" style="border-collapse: collapse" bordercolor="#111111" width="100%" id="AutoNumber1" height="83">
  <tr>
    <td rowspan="4" height="83">
    <a href="https://github.com/orthofuzz/">
    <img border="0" width="150" height="134" src="titlefigure.png"></a></td>
    <td height="15"><font color="#828282">
    <font face="Garamond" color="#000000" size="6"><i>ORTHOFUZZ</font><font face="Garamond" size="6"><b><i>&nbsp;&nbsp;&nbsp;
    </b></i></font></font></td>
  </tr>
  <tr>
    <td width="82%" height="28"><font face="Garamond" color="#000000" size="5"><b>A java library for functional annotation and pathway analysis of denovo assembled 
RNA-Seq data
</b></font></td>
  </tr>
  
 <tr>
    <td width="82%" height="28"><font face="Garamond" color="#000000" size="4"><b>http://orthofuzz.github.io/</b></font></td>
  </tr>
  <tr>
    <td width="82%" height="28"><font face="Garamond" color="#000000" size="4"><b>Nikolau Group - BBMB/BCB - Iowa State University</b></font></td>
  </tr>
</table>
<hr color="#C8C8C8">
<table cellpadding="20"><tr><td>
<p>
<div align="justify">


<span lang="EN-US" style="font-size: 14.0pt; font-family: Garamond">
<ul>
<li>

                                                                   
                                                                   
  <!-- use jssor.slider.min.js instead for release -->
    <!-- jssor.slider.min.js = (jssor.js + jssor.slider.js) -->
    <script type="text/javascript" src="js/jssor.js"></script>
    <script type="text/javascript" src="js/jssor.slider.js"></script>
    <script>
        jssor_slider1_starter = function (containerId) {
            var options = {
                $DragOrientation: 3,                                //[Optional] Orientation to drag slide, 0 no drag, 1 horizental, 2 vertical, 3 either, default value is 1 (Note that the $DragOrientation should be the same as $PlayOrientation when $DisplayPieces is greater than 1, or parking position is not 0)
                $ArrowNavigatorOptions: {                       //[Optional] Options to specify and enable arrow navigator or not
                    $Class: $JssorArrowNavigator$,              //[Requried] Class to create arrow navigator instance
                    $ChanceToShow: 2,                               //[Required] 0 Never, 1 Mouse Over, 2 Always
                    $AutoCenter: 0,                                 //[Optional] Auto center arrows in parent container, 0 No, 1 Horizontal, 2 Vertical, 3 Both, default value is 0
                    $Steps: 1                                       //[Optional] Steps to go for each navigation request, default value is 1
                }
            };

            var jssor_slider1 = new $JssorSlider$(containerId, options);
        };
    </script>
    <!-- Jssor Slider Begin -->
    <!-- To move inline styles to css file/block, please specify a class name for each element. --> 
    <div id="slider1_container" style="position: relative; top: 0px; left: 110px; width: 1000px;
        height: 450px; ">

        <!-- Slides Container -->
        <div u="slides" style="cursor: move; position: absolute; left: 0px; top: 0px; width: 1000px; height: 450px;
            overflow: hidden;">
            <div><img u="image" src="images/results/CombinedImage.png" /></div>
            <div><img u="image" src="images/results/Combined.Example2.png" /></div>
             <div><img u="image" src="images/results/CombinedImage.png" /></div>
            <div><img u="image" src="images/results/Combined.Example2.png" /></div>
            <div><img u="image" src="images/results/CombinedImage.png" /></div>
            <div><img u="image" src="images/results/Combined.Example2.png" /></div>
           
        </div>
        
        <!--#region Arrow Navigator Skin Begin -->
        <!-- Help: http://www.jssor.com/development/slider-with-arrow-navigator-jquery.html -->
        <style>
            /* jssor slider arrow navigator skin 03 css */
            /*
            .jssora03l                  (normal)
            .jssora03r                  (normal)
            .jssora03l:hover            (normal mouseover)
            .jssora03r:hover            (normal mouseover)
            .jssora03l.jssora03ldn      (mousedown)
            .jssora03r.jssora03rdn      (mousedown)
            */
            .jssora03l, .jssora03r {
                display: block;
                position: absolute;
                /* size of arrow element */
                width: 55px;
                height: 55px;
                cursor: pointer;
                background: url(images/a03.png) no-repeat;
                overflow: hidden;
            }
            .jssora03l { background-position: -3px -33px; }
            .jssora03r { background-position: -63px -33px; }
            .jssora03l:hover { background-position: -123px -33px; }
            .jssora03r:hover { background-position: -183px -33px; }
            .jssora03l.jssora03ldn { background-position: -243px -33px; }
            .jssora03r.jssora03rdn { background-position: -303px -33px; }
        </style>
        <!-- Arrow Left -->
        <span u="arrowleft" class="jssora03l" style="top: 123px; left: 8px;">
        </span>
        <!-- Arrow Right -->
        <span u="arrowright" class="jssora03r" style="top: 123px; right: 8px;">
        </span>
        <!--#endregion Arrow Navigator Skin End -->
        <a style="display: none" href="http://www.jssor.com">Bootstrap Slider</a>
        <!-- Trigger -->
        <script>
            jssor_slider1_starter('slider1_container');
        </script>
    </div>
    <!-- Jssor Slider End -->                                                                  
                                                                   
                                                                   
                                                                   
                                                                   
                                                                   
         
                                                                                                               
                                                                                                               
<p><b>Key terms used in the descriptions:</b> </p>

<p><b>source organisms</b>: organisms, the annotations of which are used as sources of functional annotations.
<b>source sequences</b>: sequences from the source organisms.
<b>target organism</b>: the denovo sequenced organism, which we are analyzing.
<b>target sequences</b>: sequences from target organisms.
<b>G</b>: The All-vs-All Pairwise Sequence Similarity Network.
<b>Node</b>: A sequence in the pairwise-sequence similarity network.

</p>                                                          
                                                                   
                                                                   
                                                                   
                                                                   
                                                                   
                                                                   
                                                                   
                                                                   
                                                                   
                                                                   
 <p align="center"><b>                                               
[CLICK ON TEXT IN <span style="color:#0000CC">BLUE</span> TO GET DETAILED DESCRIPTIONS AND DOWNLOAD LINKS]</b></p>                                                           
<p><b>Overview:</b></p>

<p> The short read RNA-Seq data generated by second generation sequencers like Illumna GA IIE and Illumina HiSeq2000 are typically assembled using denovo sequence assemblers like <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki">  Trinity denovo assembler </a> into contigs. The nature of the denovo assembler algorithms and the short reads results in  many-many mapping problem between contigs and gene-families causing uncertainty about the assignment of contigs to gene familes. This warrants imagining the contigs as nodes in a cross-organism/cross-species sequence similarity network. The entire RNA-Seq analysis pipeline downstream of assembly of contigs, read mapping and estimation of expression levels has to be modified to fit the network model. 
</p>

<p>The Orthofuzz package positions contigs assembled de-novo from RNA-Seq short read data among protein sequences obtained from phylogenetic neighbors of the target organism in a muti-organism pairwise sequence similarity network. </p><p>A set of well annotated organisms in the phylogenetic neighborhood of the target organisms are handpicked and an All vs All BLAST carried out on translated contigs and amino-acid sequences to define an All vs All pairwise sequence similarity network with the sequences forming the nodes and the BLAST bit-score used to define the edge weight (higher the bit-score, higher the homology). </p>
<p>The package includes a series of java programs which uses a combination of network topology and edge weights to annotate, analyze and summarize the RNA-Seq data.</p>


 <p align="center">
<h4>  Where does Orthofuzz fit into the RNA-Seq Bioinformatics Pipeline ?</h4>
</p>

<a href="pipeline.png">
    <p style="text-align:center"><img alt="Link not working" src="pipeline.png" width="900" height="700"/>
</a>





<p><b>List of External Packages used: </b>
<a href="http://jung.sourceforge.net"> JUNG </a><a href="https://github.com/rwl/ParallelColt"> Paralellcolt </a>
<a href="http://trove.starlight-systems.com"> Trove </a>
<a href="http://jblas.org"> Jblas </a>
<a href="http://www.paccanarolab.org/static_content/clusterone/cl1-cmdline-1.0.html"> ClusterOne </a><p>


<p><b>Output formats compatible with: </b>
<a href="http://www.cytoscape.org/cy3.html"> Cytoscape3 </a>
<a href="http://www.pathvisio.org/"> Pathvisio </a>
<a href="http://www.paccanarolab.org/clusterone/ "> ClusterOne </a>
<a href="http://genepattern.broadinstitute.org "> Genepattern </a>
</p>                                          


<p><b> Programs in the package:</b> <br/> 
(Click on names in <span style="color:#0000CC">BLUE</span> to get detailed descriptions and download links)</p>
                                            
<p><a href="RandomWalk.html"<b>1. RandomWalkwithRestart. </b> </a><br/>
<a href = "MaximalCliques.html"<b>2. Finding Representative & Maximal Cliques in the Graph. </b></a> <br/>  
<a href="transferannotations.html"<b>3. TransferAnnotations from Source Sequences to Target Target Sequences.</b></a><br/>
<a href="TransferExpression.html"<b>4. TransferExpression from Target Contigs to Gene-Sets.</b></a><br/>
<a href = "PredictInteractions.html"<b>5. Using known interactions in Source Organisms to Predict Interactions in the Target Organisms. </b></a><br/>
<a href = "GeneSetStatistics.html"<b>6. Implements methods to estimate Gene-Set Statistics. </b></a><br/>
<a href = "GeneListStatistics.html"<b>7.Implements methods to estimate Gene-List Statistics. </b></a><br/>
<a href = "Coexpression.html"<b>8. Implements methods to estimate coexpression between genes/contigs/gene-sets and estimate statistical significance. </b></a><br/>
                          
</p>                  

<p><b> Context of the programs :</b> <br/> 
                                    
<p>
                                    
Previous work has shown that functional ortholog proteins tends to share more neighbors with higher edge weight when compared to weak homologs
in All-vs-All pairwise sequence similarity networks. Edge weigted random walk with restart is a memory efficient and fast computational method for capturing both shared topological neighborhood and high edge-weight with [0-1] score, thereby making it very useful for measuring orthology between sequences and/or sets of sequences.      
</p>

<p> Given a set of query nodes and an All-vs-All pairwise sequence similarity network, <a href="RandomWalk.html"<b>RandomWalkwithRestart.jar </b></a> reports a cluster of nodes in the network, which are members of or/and homologous to the query set, provide global and organism normalized scores representing the strength of homology of the node to the query-set and statistical significance obtained by permulation test.</p>                                                   
<p>
    The query-set can have a single node or a set of nodes from the same and/or multiple organisms, whose sequences are in the network. This facilitates querying the network for homologs of gene families with one or more members from different source organisms in the network.
</p>                     
                                                                                                                
<p>The max normalized <em>orthoscore</em> [0-1] returned for each member node of the cluster is a strong indicator of functional orthology. The RandomWalkwithRestart, therefore is a very powerful tool for [1] transferring functional informations we know about the query sets to the cluster members or for [2] summarizing information we know about the cluster-members to estimate summary statistics for the query-set.
</p>

<p>
 
<a href="http://mathworld.wolfram.com/MaximalClique.html" target="_blank"> Maximal Cliques: </a> Examination of the topological structure of the sequence similarity network has revealed that the ortholog gene families tend to form <a href="http://mathworld.wolfram.com/Clique.html"> Cliques </a> in sequence similarity networks. Therefore, finding maximal/representative cliques in sequence similarity network is quite revealing in terms of identifying contig sequences that are strong members of gene families. 
</p> 
                             
<p>
The <a href="MaximalCliques.html"<b>  </b>FindMaximalCliques </a> method lists maximal and representative cliques in the network using the the algorithm proposed by Eppstein, Loffler and Strash [ELS method] for listing all maximal cliques in a sparse graph.
</p>
                   
                                                                                                                
<p><b>[1] Transferring information from query sets to cluster members: </b>
The standard approach for assigning functional annotations to denovo assembled sequences is based on the observation that there is a high probability for cross organism orthologs to share biological function. Previous work has shown that true orthologs tend to share more neighbors with high edge weights in pairwise All-vs-All pairwise sequence similarity network. Since orthoscore captures this information, we can use the orthoscores for transferring functional annotations associated with the source sequences in the networks to the target sequences.</p>

Given the source sequences and their functional annotations, <a href="transferannotations.html"<b>TransferAnnotations.jar </b></a> method will transfers functional annotations from the source sequences to the target sequences.</p> 
                                                                                               
If a gene X is known to interact with gene Y in one/many of the source organisms, <a href = "PredictInteractions.html"<b>PredictInteractions.jar </b></a> can use the RandomWalkwithRestart.jar to identify corresponding orthologs in the target organisms and predict interactions between contigs/sets of contigs which could be coding for proteins which interacts with each other in the target organisms.</p>

<p><b>[2] Transferring information from the cluster members to summarize the query sets: </b>

<p> Given mappings of gene-ids in the source organisms to enzymes, <a href="TransferExpression.html"<b>TransferExpression.jar </b></a> method will first use Randomwalkwithrestart.jar to find orthologs of the query sets in the new organisms, summarize the expression values in FPKM/TPM for each query and generate geneset statistics using <a href="GeneSetStatistics.html"> GeneSetStatistics.jar </a>  program described in the next section. </p>



<p><b>[3] Tools to statistically analyze and summarize the results: </b></p>  
<p><a href="GeneSetStatistics.html"> Gene-Set Statistics: </a> The program implements the <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3929847/" target="_blank" <b>Fisher combined probability (FCP) method </b></a> and the <a href="http://statweb.stanford.edu/~tibs/GSA/" target="_blank" <b>Gene Set Enrichment (GSA) </b></a> permutation and randomization method proposed by Bradley Efron and Rob Tibshirani. The max. mean statistic is used as the test statistic.<p>                                                     

<p><a href="GeneListStatistics.html"> Gene-List Statistics: </a> The program uses a weighted count derived from weighted annotations assigned to the target sequences by the TransferAnnotation method in the package. The <a href="https://david.ncifcrf.gov/helps/functional_annotation.html#fisher"> EASE score ( modified Fisher's Exact Test)  </a> is estimated and the statistical significance is reported.                                                 </p>
      
  <p><a href="Coexpression.html"> Coexpression: </a>A parallel implementation of coexpression from <a href="https://github.com/rwl/ParallelColt"> Paralellcolt </a> package is used for estimating Pearson's Correlation and the statistical significance for the scores are estimated by transforming the &rho; to t statistic.</p>                                                                                                 
                                                             

 
                                <p><b>User-Manual Links:</b></p>


<p><a href="ProgramManual.html"> CLICK HERE </a> for the user-manual of the programs.<p>                                       <p><a href="FileFormats.html"> CLICK HERE </a> for the file formats.<p>                                                         <p><a href-"ExampleRun.html"> CLICK HERE </a> for a Example Run </p>                        

<li><b>Source code:</b><br>

<p>The software was developed in JAVA on GNU/Linux operating system with support for multithreading. It is free for noncommercial use.</p>
<a href="javascript: void(0)" onclick="window.open('download.php','popup','width=400,height=200,scrollbars=no,resizable=no,toolbar=no,directories=no,location=no,menubar=no,status=no,left=0,top=0'); return false">link here</a>

<br><br>
<li><b>Citation:</b><br>
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21594272">Details of Paper here</a>
<br><br>
<li><b>Contact:</b><br>
Adarsh Jose (ajose<b>AT</b>iastate.edu)<br>
<font face="Garamond" color="#000000" size="3">PhD (Fall 2015), Bioinformatics and Computational Biology (BCB)</br>Iowa State University, IA</br></font><font face="Garamond" size="5">
    <font face="Garamond" color="#000000" size="4"><b><a href="https://www.linkedin.com/pub/adarsh-jose/16/858/473"> Linkedin Profile </a></b></font>
</ul>

                                             
                                             
</body></html>
