<html>

<head>
<style>
body{color:black;font-family:"Verdana", "Arial", "Helvetica", "sans-serif";font-size:9pt;margin:0}
td{color:black;font-family:"Verdana", "Arial", "Helvetica", "sans-serif";font-size:9pt}
a:link       { text-decoration: none; color:#0000FF }
a:visited    { text-decoration: none; color: #008080 }a:hover      { color: #00FFCC }
</style>
<style type="text/css">
body{margin-left:60pt;}
body{margin-right:60pt;}
</style>
<title>Orthofuzz</title>
<meta http-equiv="Content-Language" content="en-us">
<meta name="GENERATOR" content="Microsoft FrontPage 5.0">
<meta name="ProgId" content="FrontPage.Editor.Document">
<meta http-equiv="Content-Type" content="text/html; charset=windows-1252">
</head>

<body>

<p align="center">
<br>
<table border="0" cellpadding="0" cellspacing="0" style="border-collapse: collapse" bordercolor="#111111" width="100%" id="AutoNumber1" height="83">
  <tr>
    <td rowspan="4" height="83">
    <a href="http://orthofuzz.github.io/">
    <img border="0" width="150" height="134" src="titlefigure.png"></a></td>
    <td height="15"><font color="#828282">
    <font face="Garamond" color="#000000" size="6"><i>ORTHOFUZZ</font><font face="Garamond" size="6"><b><i>&nbsp;&nbsp;&nbsp;
    </b></i></font></font></td>
  </tr>
  <tr>
    <td width="82%" height="28"><font face="Garamond" color="#000000" size="5"><b>A java library for functional annotation and pathway analysis of denovo assembled 
RNA-Seq data
</b></font></td>
  </tr>
  
 <tr>
    <td width="82%" height="28"><font face="Garamond" color="#000000" size="4"><b>http://orthofuzz.github.io/</b></font></td>
  </tr>
  <tr>
    <td width="82%" height="28"><font face="Garamond" color="#000000" size="4"><b>Nikolau Group - BBMB/BCB - Iowa State University</b></font></td>
  </tr>
</table>
<hr color="#C8C8C8">
<table cellpadding="20"><tr><td>
<p>
<div align="left">


<span lang="EN-US" style="font-size: 14.0pt; font-family: Garamond">
<ul>
<li>


<p><b>Overview:</b></p>



<p>The Orthofuzz package positions contigs assembled de-novo from RNA-Seq short read data among protein sequences obtained from phylogenetic neighbors of the target organism in a muti-organism pairwise sequence similarity network using pairwise BLAST scores.</p>


 <h2>Downstream Analysis of RNA-Seq data:</h2>
<li><a> href=“titlefigure.png"
<img src=“titlefigure.png" alt="Mountain View" style="width:800px;height:600px;">
</a></li>


<p><b>Key Terms :</b> </p>

<p><b>source organisms</b>: organisms, the annotations of which are used as sources of functional annotations.
<b>source sequences</b>: sequences from the source organisms.
<b>target organism</b>: the denovo sequenced organism, which we are analyzing.
<b>target sequences</b>: sequences from target organisms.</p>
</i>

<p><b>List of External Packages used: </b></p>
<a href="http://jung.sourceforge.net"> JUNG </a><a href="https://github.com/rwl/ParallelColt"> Paralellcolt </a>
<a href="http://trove.starlight-systems.com"> Trove </a>
<a href="http://jblas.org"> Jblas </a>
<a href="http://www.paccanarolab.org/static_content/clusterone/cl1-cmdline-1.0.html"> ClusterOne </a><p>


<p><b>Compatible with: </b></p>
<a href="http://www.cytoscape.org/cy3.html"> Cytoscape3 </a><p>


<p><b> Programs in the package </b></p>


<p>A series of java programs which uses network topology and edge weights to carry out analysis of the RNA-Seq data. The package include the following interdependent java programs:</p>

<p>A set of organisms in the phylogenetic neighborhood of with assigned functional annotations are chosen and All vs All BLAST carried out on translated contigs and amino-acid sequences to define an All vs All pairwise sequence similarity network.</p>

<p>The topological neighborhood of the pairwise sequence similarity networks and BLAST bit scores are then used to transfer functional annotations from source organisms to targets, summarize gene-expression values of gene sets and predict interaction relationships between the denovo contigs.</p>

<p><b> Series of questions the orthofuzz can try to answer are listed below ! </b><p>


<p>1. Given a set of query genes in the network, <b>NetworkFlow.jar</b> will find the homologs of the queries in the network and report global and organism normalized scores representing the strength of homology.</p>

<p>2. Given the source sequences and their functional annotations, <b>TransferAnnotations.jar </b>will transfers functional annotations from the source sequences to the target sequences.</p>

<p>3. Given mappings to genes in source organisms to enzymes and pathways, <b>TransferExpressions.jar</b> will find homologs of the query sets in the new organisms, summarize gene expression values for each query. The <b>GeneSetEnrichmentStats.jar</b> will generate gene-set statistics using Fisher’s method for combining p-values from gene-sets.</p>

<p>4. Given a list of contigs, <b>GeneListStatistics.jar</b> will generate gene list statistics of lists of contigs to identify functional annotation terms enriched in lists of contigs. The enrichment statistics are estimated using EASE scores, which is a modified form of Fisher Exact Test for enrichment.</p>

<p>5. If genes in the source organisms are known to interact with each other, <b>PredictInteractions.jar </b>will find homologs of the interacting units in the target organism and report a score (0-1) estimated by finding the geometric mean of the interaction scores and the orthofuzz scores. It also summarizes the expression level of the interaction units in the target organisms.</p>

<p>6. Better strength to this prediction can be obtained by finding interaction-units that are correlated with each other. <b>ParalellCoexpression.jar </b>Estimates pearson correlation of expression data using paralellColt java package. </p>

<b>7. FindRepresentativeCliques.jar </b>finds clusters in the network, first by finding representative cliques in the network and then running NetworkFlow.jar to find weak members of the clusters.
<h4> The program parameters </h4>
<p><a href="network.flow.html"> Click here to see how to use the programs</a><p>


<p>Overview of the workflow:</p>

<p>1. Preparing the input files </p>

<p> The different input files needed for the programs have to be in <tab> delimited text files in specified formats.</p>

<p><a href="Input.formats.html"> The input file formats are detailed here. </a><p>

<p>2. Run NetworkFlow program to estimate the strength of homology between the source and target sequences.</p>

<p>3. Run the TransferExpression program to estimate the representative expression levels for query gene sets from the complied pathways.</p>

<p>4. Run the TransferAnnotations program to transfer the annotations from the source sequences to the target sequences.</p>

<p>5. Run the TransferInteractions program to estimate evidence for the presence and strength of protein interactions of the source organisms in the target organisms.</p>

<p>6. Run Coexpression program on the expression of the interaction data to identify interaction units that are co-expressed.</p>

<p>10. Run NetworkAND program to identify protein complexes that are interacting and are co-expressed in the target organisms.</p>

<p>11. Run Coexpression program on the pathway expressions to identify enzymes and pathways that are co-expressed.</p>


<p><a href="networkflow.html"> Input formats </a><p>

<p>
<br>
<br>
<br>
<br>
<br>
</p>


<p>Key concepts [Manuscript under preparation]:</p>

<p>1. The fragmented nature and the high identity between sub-groups of the contigs make the current approach of assignment of contigs to the best hits in the source organisms [Table X] error-prone [Figure X]. </p>

<p>2. Protein sequences from the same gene family share more neighbors with higher edge weights compared to those that just share functional domains [Figure X].</p>

<p>3. Random walking with restart on an edge-weighted network captures high edge weights and topological neighborhood to give probabilistic definition of network neighborhood [Figure X].</p>

<p>There are five algorithms at the core of the package [Refer Manuscript]:</p>

<p>1. Random walk with restart on an edge weighted network starting from a set of start nodes.</p>

<p>2. Estimate summary expression value of the topological neighbors of a set of nodes.</p>

<p>3. Transfer functional annotations from source organisms to target organisms.</p>

<p>4. Transfer interactions from source organisms to the target organisms.</p>

<p>5. Modified Tomita Algorithm to identify representative cliques in sequence similarity networks.</p>


<p> Statistics </p>

<p>1. Edge Weighted Random Walk with restart using perambulation statistics</p>

<p>2. Permutation Statistics using max mean statistic</p>

<p>3. EASE Score for gene list </p>

<p>4. Fishers Gene Set Stat</p>

<p>5. Max mean stat for Gene Set Analysis</p>


<p>The software was developed in JAVA on GNU/Linux operating system with support for multithreading. It is free for noncommercial use.</p>

<br><br>
<li><b>Source code:</b><br>

<a href="javascript: void(0)" onclick="window.open('download.php','popup','width=400,height=200,scrollbars=no,resizable=no,toolbar=no,directories=no,location=no,menubar=no,status=no,left=0,top=0'); return false">link here</a>

<br><br>
<li><b>Citation:</b><br>
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21594272">Details of Paper here</a>
<br><br>
<li><b>Contact:</b><br>
Adarsh Jose (ajose<b>AT</b>iastate.edu)<br>

</ul>

 <p style="line-height:150%; font-size:12pt;">
</td>
</tr></table>

</body></html>
